nltk/translate/bleu_score.py

```diff
@@ -169,11 +169,15 @@ def corpus_bleu(
         "The number of hypotheses and their reference(s) should be the " "same "
     )
 
+    if isinstance(weights, tuple):
+        weights = [weights]
```

Comment:
@BatMrE I think the functionality of using weights like `[0.2, 0.8]` like in my example above should still be supported. I would rather not merge this PR until that is resolved.

Reply:
Hey @tomaarsen,

I agree with your point however while testing for the script:
```
from nltk.translate.bleu_score import corpus_bleu
# define ref's and hyp's here, left out for conciseness.
weights = [0.2, 0.8]
bleu_scores = corpus_bleu(
    list_of_references=[[ref1a, ref1b, ref1c], [ref2a]],
    hypotheses=[hyp1, hyp2],
    weights=weights,
)
```
with code:
```
    if isinstance(weights, tuple):
        weights = [weights]
    max_weight_length = max(len(weight) for weight in weights)
```
I am getting the expected result as **0.7496592017168556**
and **_460 passed, 32 skipped, 9 xfailed, 1 warning in 222.27s (0:03:42)_**


also your code:
```
    if weights and isinstance(weights[0], float):
        max_weight_length = max(len(weight) for weight in weights)
```
gives the same result **0.7496592017168556**
and  _**12 failed, 448 passed, 32 skipped, 9 xfailed, 1 warning in 210.82s (0:03:30)**_

Reply:
I'm not sure how you get the expected result with your current implementation. I still only get
```python
TypeError: object of type 'float' has no len()
```
<details>
<summary>This is the script I'm using:</summary>

```python
hyp1 = [
    "It",
    "is",
    "a",
    "guide",
    "to",
    "action",
    "which",
    "ensures",
    "that",
    "the",
    "military",
    "always",
    "obeys",
    "the",
    "commands",
    "of",
    "the",
    "party",
]
ref1a = [
    "It",
    "is",
    "a",
    "guide",
    "to",
    "action",
    "that",
    "ensures",
    "that",
    "the",
    "military",
    "will",
    "forever",
    "heed",
    "Party",
    "commands",
]
ref1b = [
    "It",
    "is",
    "the",
    "guiding",
    "principle",
    "which",
    "guarantees",
    "the",
    "military",
    "forces",
    "always",
    "being",
    "under",
    "the",
    "command",
    "of",
    "the",
    "Party",
]
ref1c = [
    "It",
    "is",
    "the",
    "practical",
    "guide",
    "for",
    "the",
    "army",
    "always",
    "to",
    "heed",
    "the",
    "directions",
    "of",
    "the",
    "party",
]
hyp2 = [
    "he",
    "read",
    "the",
    "book",
    "because",
    "he",
    "was",
    "interested",
    "in",
    "world",
    "history",
]
ref2a = [
    "he",
    "was",
    "interested",
    "in",
    "world",
    "history",
    "because",
    "he",
    "read",
    "the",
    "book",
]

weights = [0.2, 0.8]

from nltk.translate.bleu_score import corpus_bleu

out = bleu_scores = corpus_bleu(
    list_of_references=[[ref1a, ref1b, ref1c], [ref2a]],
    hypotheses=[hyp1, hyp2],
    weights=weights,
)

print(out)
```

</details>

---

Can you explain how you get the correct results with weights `[0.2, 0.8]` currently?

Also, I meant something along the lines of 
```python
if weights and isinstance(weights[0], float):
    weights = [weights]
max_weight_length = max(len(weight) for weight in weights)
```
That said, this does give problems when the weights are for example integers, like `(0, 0, 1)`.